migrations/20160406175457_add_meta_fields_to_entities.js

```diff
@@ -0,0 +1,50 @@
+var tables = [
+    'interventions',
+    'locations',
+    'organisations',
+    'persons',
+    'problems',
+    'records',
+    'sources',
+    'trials',
+]
+
+exports.up = (knex) => {
+
+  var operations = []
+  tables.forEach((table) => {
+      var addTimestamps = knex.schema.table(table, (table) => {
+          table.timestamp('created_at').nullable();
+          table.timestamp('updated_at').nullable();
+          table.specificType('links', 'text[]').nullable();
+          table.specificType('facts', 'text[]').nullable();
+      });
+      var addLinks = knex.schema.raw(
+        'CREATE INDEX '+table+'_links_idx on '+table+' USING GIN(links)')
+      var addFacts = knex.schema.raw(
+        'CREATE INDEX '+table+'_facts_idx on '+table+' USING GIN(facts)')
```

Comment:
Instead of creating the indexes using raw SQL, you could use:

```javascript
table.specificType('links', 'text[]')
  .nullable()
  .index();
```

There's a way to specify the index type to be GIN. It should be easy, although I'm not sure how to use it (would have to test). Please note the structure of the code as well, chaining the methods in newlines...

Also, it won't be useful here, but you can use string interpolation in JavaScript ES6 as `'CREATE INDEX ${table}_links_idx ON ${table} USING GIN(links)'`